src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -116,6 +116,11 @@ func ParseRequest(
 		return ParsedOptions{}, rErr
 	}
 
+	fetchOpts.Timeout = opts.TimeoutOpts().FetchTimeout
```

Comment:
this logic is already done for you through `parseInstaneousParams`/`parseParams`. ultimately `parseParams` calls `ParseRequestTimeout`, which returns either the param or configured timeout.

I think the only thing we actually need to change is below when you use `parsed.FetchOpts.Timeout` instead of `fetchOpts.Timeout`